sufia-models/app/models/concerns/sufia/generic_file/csv.rb

```diff
@@ -0,0 +1,23 @@
+module Sufia
+  module GenericFile
+    module CSV
+      def to_csv
+        ::CSV.generate do |csv|
+          csv << csv_terms.map do |term|
+            values = send(term)
+            values = [values].flatten # make sure we have an array
+            values.join(multi_value_separator)
```

Comment:
I'd have to defer to @jcoyne there, but I should say that I wouldn't hold up the PR because of this.